File: backend/src/main/java/com/herd/squire/models/items/Tool.java
package com.herd.squire.models.items;

import com.herd.squire.models.attributes.ToolCategory;

public class Tool extends Item {
    private ToolCategory category;

    public Tool() {
    }

    public Tool(String id, String name, String description, int sid, boolean author, int version,
                boolean expendable, boolean equippable,
                EquipmentSlotType slot, boolean container, boolean ignoreWeight, int cost, CostUnit costUnit, double weight,
                ToolCategory category) {
        super(id, name, ItemType.TOOL, description, sid, author, version, expendable, equippable, slot, container, ignoreWeight, cost, costUnit, weight);
        this.category = category;
    }

    public ToolCategory getCategory() {
        return category;
    }

    public void setCategory(ToolCategory category) {
        this.category = category;
    }
}

File: frontend/src/app/shared/models/items/tool.ts
import {Item} from './item';
import {ItemType} from './item-type.enum';
import {ToolCategory} from '../attributes/tool-category';
import {ItemTypeValue} from '../../../constants';

export class Tool extends Item {
  type: ItemTypeValue = 'Tool';
  itemType: ItemType = ItemType.TOOL;
  category: ToolCategory;
}
